.github/workflows/build_packages.yml

```diff
@@ -67,18 +72,25 @@ jobs:
       - name: Build CUDA Quantum
         id: cudaq_build
         run: |
-          loaded=`docker load --input ${{ inputs.devdeps_archive }} | grep -o 'Loaded image: \S*:\S*' | cut -d ' ' -f 3`
-          devdeps_image=`echo $loaded | cut -d ":" -f 1`
-          devdeps_tag=`echo $loaded | cut -d ":" -f 2`
+          if ${{ steps.restore.outcome != 'skipped' }}; then
+            base_image=`docker load --input ${{ inputs.devdeps_archive }} | grep -o 'Loaded image: \S*:\S*' | cut -d ' ' -f 3`
+          elif ${{ inputs.devdeps_image != '' }}; then
+            docker pull ${{ inputs.devdeps_image }}
+            base_image=${{ inputs.devdeps_image }}
+          else
+            echo "Missing configuration for development dependencies. Either specify the image (i.e. provide devdeps_image) or cache (i.e. provide devdeps_cache and devdeps_archive) that should be used for the build." >> $GITHUB_STEP_SUMMARY
+            exit 1
+          fi
 
           docker build -t cuda-quantum-dev:local -f docker/build/cudaqdev.Dockerfile . \
-            --build-arg build_environment=$devdeps_image --build-arg env_tag=$devdeps_tag \
+            --build-arg base_image=$base_image \
```

Comment:
I don't think so - if this causes parsing issues here, then I think it would also cause issue in the docker file. Were you thinking of something specific?

Reply:
It seems very unlikely, but I was wondering if there was a possibility of `$base_image` expanding to something with a space in it, for example.

Reply:
> It seems very unlikely, but I was wondering if there was a possibility of $base_image expanding to something with a space in it, for example.

That would cause issues with docker. Only lower case letter, digits, periods, slashes, dashes and underscore are valid, in addition to the colon separating the tag.